### PR TITLE
Azure: enable podvm images nightly build

### PR DESCRIPTION
**Pull request by kartikjoshi21**

### CODE REVIEW

File: .github/workflows/azure-podvm-image.yml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      with:
+        # The paths to cache are documented here: https://doc.rust-lang.org/cargo/guide/cargo-home.html#caching-the-cargo-home-in-ci
+        path: |
+          ~/.cargo/.crates.toml
+          ~/.cargo/.crates2.json
+          ~/.cargo/bin/
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+        key: rust
```
[the remark this answers is not in the document]
Comment (surajssd) [reply .github/workflows/azure-podvm-image.yml:60]: I think this cache is worth it, because it saves a few double digit seconds every time those files are downloaded. And yes, we do hit this cache because they key is static. Also those recommendations are from rust upstream: 
image

File: ci-infra/azure/main.tf [hunk 2]
```diff
@@ -72,6 +72,16 @@ resource "azurerm_shared_image_gallery" "podvm_image_gallery" {
...
   resource_group_name = azurerm_resource_group.ci_rg.name
   location            = azurerm_resource_group.ci_rg.location
+
+  sharing {
+    permission = "Community"
+    community_gallery {
+	prefix = "cocopodvm"
+	eula   = "https://raw.githubusercontent.com/confidential-containers/confidential-containers/main/LICENSE"
+	publisher_uri = "https://github.com/confidential-containers/confidential-containers"
+	publisher_email = "kartikjoshi@microsoft.com"
```
Comment (surajssd) [inline ci-infra/azure/main.tf:82]: We need some generic email id isn't it?

File: ci-infra/azure/main.tf [hunk 3]
```diff
@@ -87,4 +97,5 @@ resource "azurerm_shared_image" "podvm_image" {
   }
   hyper_v_generation        = "V2"
   confidential_vm_supported = true
+
```
Comment (surajssd) [inline ci-infra/azure/main.tf:100]: Seems unrelated? 
```suggestion
```

File: ci-infra/azure/outputs.tf [hunk 4]
```diff
@@ -33,3 +33,7 @@ output "AZURE_PODVM_IMAGE_DEF_NAME" {
 output "AZURE_MANAGED_IDENTITY_NAME" {
   value = azurerm_user_assigned_identity.gh_action_user_identity.name
 }
+
+output "AZURE_COMMUNITY_GALLERY_NAME" {
+  value = azurerm_shared_image_gallery.podvm_image_gallery.sharing[0].community_gallery[0].name
+}
```
Comment (surajssd) [inline ci-infra/azure/outputs.tf:39]: We need to get these terraform changes in, before we merge the workflow. Because without the terraform changes, the workflow will publish an image but it won't be public. Which should not fail per se, but then we have to run the terraform commands before UTC midnight 😆

Comment (kartikjoshi21) [reply ci-infra/azure/outputs.tf:39]: Sooo terraform commit will get merged first if we merge this whole PR that should suffice right ?

File: ci-infra/azure/main.tf (hunk 2 above)
Comment (kartikjoshi21) [reply ci-infra/azure/main.tf:82]: Yess it should be something generic, i saw other community gallery there we are using Magnus's id so for time being i used this one. What do you suggest which one we should use?

File: ci-infra/azure/outputs.tf (hunk 4 above)
Comment (surajssd) [reply ci-infra/azure/outputs.tf:39]: I think it is ok, we can merge it together. We have a small window before the workflow runs.

File: ci-infra/azure/main.tf (hunk 2 above)
Comment (surajssd) [reply ci-infra/azure/main.tf:82]: We can use the coco email id: cocoatmsft@outlook.com

Comment (surajssd) [reply ci-infra/azure/main.tf:82]: or if @bpradipt has some suggestions on this one?

Comment (bpradipt) [reply ci-infra/azure/main.tf:82]: [cocoatmsft@outlook.com](mailto:cocoatmsft@outlook.com) looks fine.

**Review by mkulke — COMMENTED** [review]
Could we turn this into a [reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows) to share the code with the azure e2e test?

**Comment by surajssd** [comment]
> Could we turn this into a [reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows) to share the code with the azure e2e test?

We should definitely do that. We don't have to maintain code in two places for the same thing.

**Comment by mkulke** [comment]
> > Could we turn this into a [reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows) to share the code with the azure e2e test?
> 
> We should definitely do that. We don't have to maintain code in two places for the same thing.

I took the commits from this PR and consolidated it with the e2e test workflow to share the build job.

**Comment by kartikjoshi21** [comment]
Merged along with [This](https://github.com/confidential-containers/cloud-api-adaptor/pull/1391#pullrequestreview-1609021629) !!
